lumicks/pylake/kymotracker/detail/msd_estimation.py

```diff
@@ -37,6 +63,90 @@ def calculate_msd(frame_idx, position, max_lag):
     return frame_lags, msd
 
 
+def _msd_diffusion_covariance(max_lags, n, intercept, slope):
+    """Covariance matrix for the mean squared displacements.
+
+    Equation 8a from [2].
+
+    Parameters
+    ----------
+    max_lags : max_lags
+        number of lags used in estimation.
+    n : int
+        number of trajectory points used
+    intercept : float
+        estimated localization uncertainty
+    slope : float
+        estimate for the slope
+
+    2) Bullerjahn, J. T., von Bülow, S., & Hummer, G. (2020). Optimal estimates of self-diffusion
+    coefficients from molecular dynamics simulations. The Journal of Chemical Physics, 153(2),
+    024116.
+    """
+    # Intercept corresponds to a^2 in the paper, slope refers to sigma^2 in the paper
+    i = np.tile(np.arange(max_lags) + 1, (max_lags, 1))
+    j = i.T
+    min_ij = np.minimum(i, j)
+
+    # Covariance matrix for localization uncertainty = 0
+    term1 = 2.0 * min_ij * (1.0 + 3.0 * i * j - min_ij**2) / (n - min_ij + 1)
+    denominator = (n - i + 1.0) * (n - j + 1.0)
+    term2 = (min_ij**2 - min_ij**4) / denominator
+    heaviside = (i + j - n - 2) >= 0
+    term3 = heaviside * ((n + 1.0 - i - j) ** 4 - (n + 1.0 - i - j) ** 2) / denominator
+
+    # Covariance matrix if there was no localization uncertainty (Eq 8b)
+    base_covariance = (slope**2 / 3) * (term1 + term2 + term3)
+
+    # Intercept corresponds to a^2 in the paper, slope refers to sigma^2 in the paper
+    term4_nominator = intercept**2 * (1.0 + (i == j)) + 4 * intercept * slope * min_ij
```

Comment:
numerator

Reply:
Oh man, that word has been a total blind spot for me. I've consistently been calling that thing nominator 😊